templates/CRM/common/paymentBlock.tpl

```diff
@@ -111,15 +114,16 @@
           }
         }
       }
-      dataUrl =  dataUrl + "&payment_instrument_id=" + payment_instrument_id;
+      dataUrl =  dataUrl + "&payment_instrument_id=" + payment_instrument_id + "&currency=" + currency;
 
       // Processors like pp-express will hide the form submit buttons, so re-show them when switching
       $('.crm-submit-buttons', $form).show().find('input').prop('disabled', true);
       CRM.loadPage(dataUrl, {target: '#billing-payment-block'});
     }
 
-    $('[name=payment_processor_id]').on('change.paymentBlock', function() {
-        buildPaymentBlock($(this).val());
+    $('[name=payment_processor_id], #currency').on('change.paymentBlock', function() {
+      var payment_processor_id = $('[name=payment_processor_id]:checked').val() == undefined ? $('[name=payment_processor_id]').val() : $('[name=payment_processor_id]:checked').val();
+      buildPaymentBlock(payment_processor_id);
```

Comment:
@monishdeb This breaks backend New Contribution forms without a payment processor (ex: Contact Record -> New Contribution, then change the currency).

Adding "if (payment_contribution_id) { .. }" fixes it, although I haven't tested extensively.

For reference, this form:

![civi-2020-11-06_13-44](https://user-images.githubusercontent.com/254741/98403027-47cef380-2036-11eb-97e9-974ddfd7cdec.png)